File: LinkedList/1D LinkedList/Reverse Between/reverse.go
type ListNode struct {
    Val  int
    Next *ListNode
}

func reverseBetween(head *ListNode, left int, right int) *ListNode {
    if head == nil || left == right {
        return head
    }

    dummy := &ListNode{Val: 0, Next: head}
    prev := dummy

    // Move prev one node before left
    for i := 0; i < left-1; i++ {
        prev = prev.Next
    }

    cur := prev.Next

    // In-place reversal of sublist
    for i := 0; i < right-left; i++ {
        temp := cur.Next
        cur.Next = temp.Next
        temp.Next = prev.Next
        prev.Next = temp
    }

    return dummy.Next
}

File: LinkedList/1D LinkedList/Reverse Between/reverse.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseBetween(self, head: ListNode, left: int, right: int) -> ListNode:
        if not head or left == right:
            return head

        dummy = ListNode(0)
        dummy.next = head
        prev = dummy

        # Move prev one node before 'left'
        for _ in range(left - 1):
            prev = prev.next

        cur = prev.next

        # Reverse sublist between left and right
        for _ in range(right - left):
            temp = cur.next
            cur.next = temp.next
            temp.next = prev.next
            prev.next = temp

        return dummy.next
